src/Handler/Image/CoreType.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Bolt\Filesystem\Handler\Image;
+
+/**
+ * A core (built-in) image type.
+ *
+ * @author Carson Full <carsonfull@gmail.com>
+ */
+final class CoreType implements TypeInterface
+{
+    /** @var int */
+    private $id;
+    /** @var string */
+    private $name;
+
+    /**
+     * Returns a list of all the image types.
+     *
+     * @return TypeInterface[]
+     */
+    public static function getTypes()
+    {
+        $types = [];
+        foreach (static::getConstants() as $id => $name) {
+            $types[] = new static($id, $name);
+        }
+
+        return $types;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param int    $id   An IMAGETYPE_* constant
+     * @param string $name String representation based on constant
+     */
+    private function __construct($id, $name)
+    {
+        $this->id = (int) $id;
+        $this->name = $name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMimeType()
+    {
+        return image_type_to_mime_type($this->id);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getExtension($includeDot = true)
+    {
+        return image_type_to_extension($this->id, $includeDot);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toString()
+    {
+        return $this->name;
+    }
```

Comment:
So a magic method to call a method to access a class object? You've rejected my PRs for less :stuck_out_tongue_winking_eye: